library/core/class.markdownvanilla.php

```diff
@@ -224,4 +224,54 @@ protected function makeCodeSpan($code) {
         $code = str_replace(["\r", "\n"], ' ', $code);
         return parent::makeCodeSpan($code);
     }
+
+    /**
+     * Main function. Performs some preprocessing on the input text and pass
+     * it through the document gamut.
+     *
+     * @api
+     *
+     * @param  string $text
+     * @return string
+     */
+    public function transform($text) {
```

Comment:
Overriding the main method make me nervous. Can you document the changes you made better here?